types/share_splitting.go

```diff
@@ -113,7 +113,13 @@ func getNextChunk(rawDatas [][]byte, outerIndex int, innerIndex int, width int)
 func GenerateTailPaddingShares(n int, shareWidth int) NamespacedShares {
 	shares := make([]NamespacedShare, n)
 	for i := 0; i < n; i++ {
-		shares[i] = NamespacedShare{bytes.Repeat([]byte{0}, shareWidth), consts.TailPaddingNamespaceID}
+		shares[i] = NamespacedShare{
+			Share: append(
+				append(make([]byte, 0, shareWidth), consts.TailPaddingNamespaceID...),
+				bytes.Repeat([]byte{0}, shareWidth-consts.NamespaceSize)...,
```

Comment:
Would it make sense to move the `bytes.Repeat([]byte{0}, shareWidth-consts.NamespaceSize)` out of the for-loop? The compiler is probably smart enough to optimize this anyways but it would be slightly more readable with a proper var name imo.

Reply:
Actually, I think we can move out not just repeat but the whole slice in global var to be allocated once and reused everywhere. Shares are immutable so this should work and save us some RAM 😌 

Furthermore, using a pointer for NamespaceShare instead of copying value should also give us a tiny win, though still a win. But this should be tackled together with whole Share struct revisiting, as it currently has an ID field which is totally unused, but still exist